### PR TITLE
Pass an error when Cosmos DB returns error response

### PR DESCRIPTION
**Pull request by jaelim-ms**
* [ ] Has `CHANGELOG.md` been updated?
* [X] Are tests passing locally?
* [X] Are the files formatted correctly?
* [X] Did you add unit tests?
* [X] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

HTTP request to Cosmos DB returns an error code without an error object which causes unexpected process in Data module. Add code to instantiate an error object with details in this case.

### CODE REVIEW

File: AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m [hunk 1]
```diff
@@ -469,7 +471,22 @@ - (void)performCosmosDbOperationWithPartition:(NSString *)partition
                                                                               httpMethod:httpMethod
                                                                                     body:body
                                                                        additionalHeaders:additionalHeaders
-                                                                       completionHandler:completionHandler];
+                                                                       completionHandler:^(NSData *_Nullable data,
+                                                                                           NSHTTPURLResponse *_Nullable response,
+                                                                                           NSError *_Nullable cosmosDbError) {
+                                                                         if (!cosmosDbError &&
+                                                                             ![MSHttpUtil isSuccessStatusCode:response.statusCode]) {
```
Comment (guperrot) [inline AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m]: I see only 1 new test but for 4 code branches here.

Comment (jaelim-ms) [reply AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m]: All other branches are covered by other tests already.
This is handling edge case and the unit test tests the edge case.

Comment (guperrot) [reply AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m]: ok

File: AppCenterDataStorage/AppCenterDataStorageTests/MSDataStoreTests.m [hunk 2]
```diff
@@ -520,6 +519,55 @@ - (void)testPerformCosmosDbAsyncOperationWithHttpClient {
...
+                                completionHandler:^(NSData *_Nullable responseBody, NSHTTPURLResponse *_Nullable response,
+                                                    NSError *_Nullable error) {
+                                  XCTAssertNil(responseBody);
+                                  actualResponse = response;
+                                  actualError = error;
+                                  [expectation fulfill];
+                                }];
+
+  // Then
+  [self waitForExpectationsWithTimeout:100
```
Comment (yochaude) [inline AppCenterDataStorage/AppCenterDataStorageTests/MSDataStoreTests.m]: Maybe use less than 100 here so the tests does not hang too long when it breaks 😬 ?

Comment (jaelim-ms) [reply AppCenterDataStorage/AppCenterDataStorageTests/MSDataStoreTests.m]: Oops.. That was for my test. 😓
